From 24f24e03793c8214a5d1f3414a5aeb48eccef4f4 Mon Sep 17 00:00:00 2001
From: Bhushan Shah <bhush94@gmail.com>
Date: Wed, 25 Mar 2015 21:25:22 +0530
Subject: [PATCH 4/4] Workaround the lockscreen password field focus issue

Fixes bug 344823
Backport of a476e1b6bf6f683bd74000bb30076868c9f92371 in 5.2

CCMAIL: release-team@kde.org
---
 lookandfeel/contents/lockscreen/LockScreen.qml | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lookandfeel/contents/lockscreen/LockScreen.qml b/lookandfeel/contents/lockscreen/LockScreen.qml
index 8b01322..ab554bd 100644
--- a/lookandfeel/contents/lockscreen/LockScreen.qml
+++ b/lookandfeel/contents/lockscreen/LockScreen.qml
@@ -156,6 +156,14 @@ Image {
                             enabled: !authenticator.graceLocked
                             onAccepted: unlockFunction()
                             focus: true
+                            //HACK: Similar hack is needed in sddm loginscreen
+                            //TODO: investigate
+                            Timer {
+                                interval: 200
+                                running: true
+                                repeat: false
+                                onTriggered: passwordInput.forceActiveFocus()
+                            }
                             visible: block.mainItem.model.get(block.mainItem.selectedIndex)["showPassword"]
                             onVisibleChanged: {
                                 if (visible) {
-- 
1.9.3

